models/migrations/v226.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/models/webhook"
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/secret"
+	"code.gitea.io/gitea/modules/setting"
+	api "code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/modules/timeutil"
+	"xorm.io/builder"
+	"xorm.io/xorm"
+)
+
+func addHeaderAuthorizationEncryptedColWebhook(x *xorm.Engine) error {
+	// Add the column to the table
+	type Webhook struct {
+		ID                 int64 `xorm:"pk autoincr"`
+		RepoID             int64 `xorm:"INDEX"` // An ID of 0 indicates either a default or system webhook
+		OrgID              int64 `xorm:"INDEX"`
+		IsSystemWebhook    bool
+		URL                string `xorm:"url TEXT"`
+		HTTPMethod         string `xorm:"http_method"`
+		ContentType        webhook.HookContentType
+		Secret             string `xorm:"TEXT"`
+		Events             string `xorm:"TEXT"`
+		*webhook.HookEvent `xorm:"-"`
+		IsActive           bool               `xorm:"INDEX"`
+		Type               webhook.HookType   `xorm:"VARCHAR(16) 'type'"`
+		Meta               string             `xorm:"TEXT"` // store hook-specific attributes
+		LastStatus         webhook.HookStatus // Last delivery status
+
+		// HeaderAuthorizationEncrypted should be accessed using HeaderAuthorization() and SetHeaderAuthorization()
+		HeaderAuthorizationEncrypted string `xorm:"TEXT"`
+
+		CreatedUnix timeutil.TimeStamp `xorm:"INDEX created"`
+		UpdatedUnix timeutil.TimeStamp `xorm:"INDEX updated"`
+	}
+	err := x.Sync2(new(Webhook))
+	if err != nil {
+		return err
+	}
+
+	// Migrate the matrix webhooks
+
+	type MatrixMeta struct {
+		HomeserverURL string `json:"homeserver_url"`
+		Room          string `json:"room_id"`
+		MessageType   int    `json:"message_type"`
+	}
+	type MatrixMetaWithAccessToken struct {
+		MatrixMeta
+		AccessToken string `json:"access_token"`
+	}
+
+	var start int
+	hooks := make([]*Webhook, 0, 50)
+	for {
+		err := x.Where("type=?", "matrix").OrderBy("id").Limit(50, start).Find(&hooks)
+		if err != nil {
+			return err
+		}
+
```

Comment:
Checking `if len(hooks) == 0` and break here is better, otherwise there will possibly be a blank commit.

Reply:
Changed to `if err != nil || len(hooks) == 0 {` to prevent such a blank commit and report any error that could happen (in case some hooks are returned as well as an error, I think it is important not to discard the error, which could prevent some hooks from being migrated).